admin/attributes_controller.php

```diff
@@ -146,113 +146,64 @@
     case 'set_flag_attributes_display_only':
       if (isset($_POST['divertClickProto'])) {
         $action = '';
-        if ($_GET['flag'] == '0') {
-          $db->Execute("UPDATE " . TABLE_PRODUCTS_ATTRIBUTES . "
-                        SET attributes_display_only = 1
-                        WHERE products_id = " . (int)$_GET['products_filter'] . "
-                        AND products_attributes_id = " . (int)$_GET['attributes_id']);
-        } else {
-          $db->Execute("UPDATE " . TABLE_PRODUCTS_ATTRIBUTES . "
-                        SET attributes_display_only = 0
-                        WHERE products_id = " . (int)$_GET['products_filter'] . "
-                        AND products_attributes_id = " . (int)$_GET['attributes_id']);
-        }
+        $db->Execute("UPDATE " . TABLE_PRODUCTS_ATTRIBUTES . "
+                      SET attributes_display_only = " . ($_GET['flag'] == '0' ? '1' : '0') . "
+                      WHERE products_id = " . (int)$_GET['products_filter'] . "
+                      AND products_attributes_id = " . (int)$_GET['attributes_id']);
       }
       break;
 
     case 'set_flag_product_attribute_is_free':
       if (isset($_POST['divertClickProto'])) {
         $action = '';
-        if ($_GET['flag'] == '0') {
-          $db->Execute("UPDATE " . TABLE_PRODUCTS_ATTRIBUTES . "
-                        SET product_attribute_is_free = 1
-                        WHERE products_id = " . (int)$_GET['products_filter'] . "
-                        AND products_attributes_id = " . (int)$_GET['attributes_id']);
-        } else {
-          $db->Execute("UPDATE " . TABLE_PRODUCTS_ATTRIBUTES . "
-                        SET product_attribute_is_free = 0
-                        WHERE products_id = " . (int)$_GET['products_filter'] . "
-                        AND products_attributes_id = " . (int)$_GET['attributes_id']);
-        }
-        zen_redirect(zen_href_link(FILENAME_ATTRIBUTES_CONTROLLER, $_SESSION['page_info'] . '&products_filter=' . $_GET['products_filter'] . '&current_category_id=' . $_GET['current_category_id']));

```

Comment:
Did you really intend to remove all these `zen_redirect` calls?

Reply:
Yes it is.
The first case was already missing a redirect. The others had a redirect with minimum get values. Since the the cases are entered by clicking a link, all the info is already passed to url, and not really needed.
You could use the redirect to empty the action value.

Reply:
After testing on a live site, I concluded that te redirect should be put back. It is necessary for the action Get value to empty